datajoint/fetch.py

```diff
@@ -182,6 +188,11 @@ def __call__(self, *attrs, offset=None, limit=None, order_by=None, format=None,
                 ret = np.array(ret, dtype=heading.as_dtype)
                 for name in heading:
                     ret[name] = list(map(partial(get, heading[name]), ret[name]))
+
+                    if isinstance(ret[name],(np.ndarray)) and len(ret[name]) > 0:
+                        main_type = type(ret[name][0])
+                        if all(isinstance(i, main_type) for i in ret[name]) and not isinstance(ret[name][0],str):
```

Comment:
could check for `str` before all the other checks for better efficiency.

Reply:
I agree. I have now moved the operation to prior to initializing the np.array.